proposals/named-tuples.md

```diff
@@ -0,0 +1,112 @@
+
+# Tuples: Small to Big
+
+When tuples were added in C# 7 they brought a few new things to the language
+
+   1. A short, simple syntax for grouping pieces of data
+   2. A way to group simple data beyond method boundaries with automatic structural equality
+   3. First-class support for pattern matching and destructuring with positional semantics
+
+In retrospect, it looks like tuples have done a very good job of addressing these pain points
+for small sets of data across short pieces of a program. This is great, but it presents a
+problem when you decide that your set of data has either grown too large to represent in a tuple,
+or you decide that you want that tuple to be a core type in large parts of your program. Since
+a tuple is not really its own type and instead is simply a composite of its members, this forces
+you to repeat the type declaration at every reference point, which quickly becomes laborious if
+you either have many tuple members or you use the tuple in many places. In addition, the more
+the tuple type becomes central to the design of a piece of your program, the more the structural
+typing becomes a problem and the desire for traditional C# nominal typing becomes prominent.
+
+This is generally the point where most users will want to evolve their tuple type into a proper
+named type. Unfortunately, this is extremely laborious. Now the following must be defined manually:
+
+  1. `ItemX` properties for each positional member of the type
+  2. Optionally, custom named properties for each positional member of the type
+  3. A constructor to assign each of the input parameters
+  4. A deconstructor to support pattern matching
+  4. Memberwise equality, including `GetHashCode` and `IEquatable`
```

Comment:
Note: i would be clear that while an IDE can help out with all of that (and it's something i'm working on), it's still *conceptually* very laborious.  It's just more code to see.  It's places where bugs can creep in.  It's something you need to maintain as you evolve your type.